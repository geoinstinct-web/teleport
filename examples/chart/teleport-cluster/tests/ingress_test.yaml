suite: Proxy Ingress
templates:
  - proxy/ingress.yaml
tests:
  - it: does not create an Ingress by default
    set:
      clusterName: teleport.example.com
    asserts:
      - hasDocuments:
          count: 0

  - it: creates an Ingress when ingress.enabled=true
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress

  - it: wears annotations when set
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
      annotations:
        ingress:
          test-annotation: test-annotation-value
          another-annotation: some-other-value
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress
      - equal:
          path: metadata.annotations.test-annotation
          value: test-annotation-value
      - equal:
          path: metadata.annotations.another-annotation
          value: some-other-value

  - it: sets the clusterName and wildcard of clusterName as hostnames when Ingress is enabled
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "teleport.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "teleport.example.com"
              - "*.teleport.example.com"
      - equal:
          path: spec.rules[0].host
          value: "teleport.example.com"
      - contains:
          path: spec.rules
          content:
            host: "teleport.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - equal:
          path: spec.rules[1].host
          value: "*.teleport.example.com"
      - contains:
          path: spec.rules
          content:
            host: "*.teleport.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: does not set a wildcard of clusterName as a hostname when Ingress is enabled and ingress.suppressAutomaticWildcards is true
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
        suppressAutomaticWildcards: true
      proxyListenerMode: multiplex
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "teleport.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "teleport.example.com"
      - equal:
          path: spec.rules[0].host
          value: "teleport.example.com"
      - contains:
          path: spec.rules
          content:
            host: "teleport.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - notContains:
          path: spec.rules
          content:
            host: "*.teleport.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: sets the publicAddr and wildcard of publicAddr as hostnames when Ingress is enabled and publicAddr is set
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
      publicAddr: ["helm-lint.example.com"]
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "helm-lint.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "helm-lint.example.com"
              - "*.helm-lint.example.com"
      - equal:
          path: spec.rules[0].host
          value: helm-lint.example.com
      - contains:
          path: spec.rules
          content:
            host: "helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - equal:
          path: spec.rules[1].host
          value: "*.helm-lint.example.com"
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: does not set a wildcard of publicAddr as a hostname when Ingress is enabled, publicAddr is set and ingress.suppressAutomaticWildcards is true
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
        suppressAutomaticWildcards: true
      proxyListenerMode: multiplex
      publicAddr: ["helm-lint.example.com"]
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "helm-lint.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "helm-lint.example.com"
      - equal:
          path: spec.rules[0].host
          value: helm-lint.example.com
      - contains:
          path: spec.rules
          content:
            host: "helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - notContains:
          path: spec.rules
          content:
            host: "*.helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: trims ports from publicAddr and uses it as the hostname when Ingress is enabled and publicAddr is set
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
      publicAddr: ["helm-lint.example.com:443"]
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "helm-lint.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "helm-lint.example.com"
              - "*.helm-lint.example.com"
      - equal:
          path: spec.rules[0].host
          value: "helm-lint.example.com"
      - contains:
          path: spec.rules
          content:
            host: helm-lint.example.com
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - equal:
          path: spec.rules[1].host
          value: "*.helm-lint.example.com"
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: exposes all publicAddrs and wildcard publicAddrs as hostnames when Ingress is enabled and multiple publicAddrs are set
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
      publicAddr: ["helm-lint.example.com", "helm-lint-second-domain.example.com"]
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "helm-lint.example.com"
      - equal:
          path: spec.tls[0].hosts[1]
          value: "helm-lint-second-domain.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "helm-lint.example.com"
              - "helm-lint-second-domain.example.com"
              - "*.helm-lint.example.com"
              - "*.helm-lint-second-domain.example.com"
      - equal:
          path: spec.rules[0].host
          value: "helm-lint.example.com"
      - equal:
          path: spec.rules[1].host
          value: "helm-lint-second-domain.example.com"
      - equal:
          path: spec.rules[2].host
          value: "*.helm-lint.example.com"
      - equal:
          path: spec.rules[3].host
          value: "*.helm-lint-second-domain.example.com"
      - contains:
          path: spec.rules
          content:
            host: "helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "helm-lint-second-domain.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint-second-domain.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  # this is a very contrived example which wouldn't even work in reality
  # it's just to test the logic in the hostname generation code
  - it: does not add additional wildcard publicAddrs when Ingress is enabled and a publicAddr already contains a wildcard
    set:
      clusterName: teleport.example.com
      ingress:
        enabled: true
      proxyListenerMode: multiplex
      publicAddr: ["helm-lint.example.com", "*.helm-lint.example.com", "helm-lint-second-domain.example.com:443"]
    asserts:
      - equal:
          path: spec.tls[0].hosts[0]
          value: "helm-lint.example.com"
      - equal:
          path: spec.tls[0].hosts[1]
          value: "*.helm-lint.example.com"
      - equal:
          path: spec.tls[0].hosts[2]
          value: "helm-lint-second-domain.example.com"
      - equal:
          path: spec.tls[0].hosts[3]
          value: "*.helm-lint-second-domain.example.com"
      - contains:
          path: spec.tls
          content:
            hosts:
              - "helm-lint.example.com"
              - "*.helm-lint.example.com"
              - "helm-lint-second-domain.example.com"
              - "*.helm-lint-second-domain.example.com"
      - equal:
          path: spec.rules[0].host
          value: "helm-lint.example.com"
      - equal:
          path: spec.rules[1].host
          value: "*.helm-lint.example.com"
      - equal:
          path: spec.rules[2].host
          value: "helm-lint-second-domain.example.com"
      - equal:
          path: spec.rules[3].host
          value: "*.helm-lint-second-domain.example.com"
      - contains:
          path: spec.rules
          content:
            host: "helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "helm-lint-second-domain.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - contains:
          path: spec.rules
          content:
            host: "*.helm-lint-second-domain.example.com"
            http:
              paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 443
                path: /
                pathType: Prefix
      - matchSnapshot:
          path: spec.tls

  - it: sets spec when passed
    set:
      clusterName: teleport.example.com
      proxyListenerMode: multiplex
      ingress:
        enabled: true
        spec:
          ingressClassName: nginx
          otherSpecStuff: lint
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress
      - equal:
          path: spec.ingressClassName
          value: nginx
      - equal:
          path: spec.otherSpecStuff
          value: lint
