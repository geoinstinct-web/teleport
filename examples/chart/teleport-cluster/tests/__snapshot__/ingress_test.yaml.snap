does not add additional wildcard publicAddrs when Ingress is enabled and a publicAddr already contains a wildcard:
  1: |
    - hosts:
      - helm-lint.example.com
      - '*.helm-lint.example.com'
      - helm-lint-second-domain.example.com
      - '*.helm-lint-second-domain.example.com'
does not set a wildcard of clusterName as a hostname when Ingress is enabled and ingress.suppressAutomaticWildcards is true:
  1: |
    - hosts:
      - teleport.example.com
? does not set a wildcard of publicAddr as a hostname when Ingress is enabled, publicAddr
  is set and ingress.suppressAutomaticWildcards is true
: 1: |
    - hosts:
      - helm-lint.example.com
exposes all publicAddrs and wildcard publicAddrs as hostnames when Ingress is enabled and multiple publicAddrs are set:
  1: |
    - hosts:
      - helm-lint.example.com
      - helm-lint-second-domain.example.com
      - '*.helm-lint.example.com'
      - '*.helm-lint-second-domain.example.com'
sets the clusterName and wildcard of clusterName as hostnames when Ingress is enabled:
  1: |
    - hosts:
      - teleport.example.com
      - '*.teleport.example.com'
sets the publicAddr and wildcard of publicAddr as hostnames when Ingress is enabled and publicAddr is set:
  1: |
    - hosts:
      - helm-lint.example.com
      - '*.helm-lint.example.com'
trims ports from publicAddr and uses it as the hostname when Ingress is enabled and publicAddr is set:
  1: |
    - hosts:
      - helm-lint.example.com
      - '*.helm-lint.example.com'
